Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+SwiftThunkPrinting.swift

```diff
@@ -120,6 +120,64 @@ extension JNISwift2JavaGenerator {
     printDestroyFunctionThunk(&printer, type)
   }
 
+  private func printEnumCase(_ printer: inout CodePrinter, _ enumCase: ImportedEnumCase) {
+    guard let translatedCase = self.translatedEnumCase(for: enumCase) else {
+      return
+    }
+
+    // Print static case initializer
+    printSwiftFunctionThunk(&printer, enumCase.caseFunction)
+    printer.println()
+
+    // Print getAsCase method
+    if !enumCase.parameters.isEmpty {
+      let selfParameter = JavaParameter(name: "self", type: .long)
+      let resultType = JavaType.class(package: javaPackage, name: "\(translatedCase.enumName).\(translatedCase.name)")
+      printCDecl(
+        &printer,
+        javaMethodName: "$getAs\(translatedCase.name)",
+        parentName: "\(translatedCase.enumName)",
+        parameters: [selfParameter],
+        resultType: resultType
+      ) { printer in
+        let selfPointer = self.printSelfJLongToUnsafeMutablePointer(
+          &printer,
+          swiftParentName: enumCase.enumType.nominalTypeDecl.name,
+          selfParameter
+        )
+        let caseNames = enumCase.parameters.enumerated().map { idx, parameter in
+          parameter.name ?? "_\(idx)"
+        }
+        let caseNamesWithLet = caseNames.map { "let \($0)" }
+        let methodSignature = MethodSignature(resultType: resultType, parameterTypes: translatedCase.conversions.map(\.native.javaType))
+        // TODO: Caching of class and static method ID.
+        printer.print(
+        """
+        guard case .\(enumCase.name)(\(caseNamesWithLet.joined(separator: ", "))) = \(selfPointer).pointee else {
+          fatalError("Expected enum case '\(enumCase.name)', but was '\\(self$.pointee)'!")
+        }
+        let recordClass$ = environment.interface.FindClass(environment, "\(javaPackagePath)/\(translatedCase.enumName)$\(translatedCase.name)")!
+        let fromJNIID$ = environment.interface.GetStaticMethodID(environment, recordClass$, "fromJNI", "\(methodSignature.mangledName)")!
```

Comment:
let's leave a todo somewhere about caching the method ids and class ref?

Reply:
I plan to tackle the caching in this PR as well, it is quite a significant performance boost

Reply:
Very nice, looking forward to it 👍